File: Algorithm/Single-Number-Leetcode/single_number.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
	vector<int> singleNumber(vector<int>& nums) {
		
		int xy = 0;
		for(int n: nums) xy ^= n;
		xy &= -xy;
		vector<int> result = {0, 0};
		for(int n: nums){
			if(xy & n) result[0] ^= n;
			else result[1] ^= n;
		}
		return result;
		
	}
};

int main(){
	vector<int> v1 = {1,2,1,3,2,5};
	Solution s;
	auto result = s.singleNumber(v1);
	
	cout << "Output..." << endl;
	
	for (int i: result){
		cout << i << endl;
	}
	
	return 0;
}

File: Algorithm/Single-Number-III-Leetcode/single_number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:


        result = [0]*2

        ab = 0 

        for i in nums:
            ab ^= i
            
        ab &= -ab

        for i in nums:
            if (ab & i) == 0:
                result[0] ^= i
            else:
                result[1] ^= i
                
        return result
        
